File: net/Azure.Test.PerfStress/PerfStressTest.cs
using System.Threading;
using System.Threading.Tasks;

namespace Azure.Test.PerfStress
{
    public abstract class PerfStressTest<TOptions> : IPerfStressTest where TOptions : PerfStressOptions
    {
        protected TOptions Options { get; private set; }

        public PerfStressTest(TOptions options)
        {
            Options = options;
        }

        public virtual Task GlobalSetup()
        {
            return Task.CompletedTask;
        }

        public virtual Task Setup()
        {
            return Task.CompletedTask;
        }

        public abstract void Run(CancellationToken cancellationToken);

        public abstract Task RunAsync(CancellationToken cancellationToken);

        public virtual Task Cleanup()
        {
            return Task.CompletedTask;
        }

        public virtual Task GlobalCleanup()
        {
            return Task.CompletedTask;
        }
    }
}

File: java/src/main/java/com/microsoft/storageperf/core/PerfStressTest.java
package com.microsoft.storageperf.core;

public abstract class PerfStressTest<TOptions extends PerfStressOptions> {
    protected final TOptions Options;

    public PerfStressTest(TOptions options) {
        Options = options;
    }

    public void GlobalSetup() {
    }

    public void Setup() {
    }

    public abstract void Run();

    public void Cleanup() {
    }

    public void GlobalCleanup() {
    }
}
